share-coordinator/src/main/java/org/apache/kafka/coordinator/share/ShareCoordinatorShard.java

```diff
@@ -493,6 +496,56 @@ private List<PersisterStateBatch> mergeBatches(
             .combineStateBatches();
     }
 
+    /**
+     * This method finds the ShareSnapshotValue record corresponding to the requested topic partition from the
+     * in-memory state of coordinator shard, the shareStateMap.
+     * <p>
+     * This method as called by the ShareCoordinatorService will be provided with
+     * the request data which covers only key i.e. group1:topic1:partition1. The implementation
+     * below was done keeping this in mind.
+     *
+     * @param request - ReadShareGroupStateSummaryRequestData for a single key
+     * @param offset  - offset to read from the __share_group_state topic partition
+     * @return CoordinatorResult(records, response)
+     */
+    public ReadShareGroupStateSummaryResponseData readStateSummary(ReadShareGroupStateSummaryRequestData request, Long offset) {
+        // records to read (with the key of snapshot type), response to caller
+        // only one key will be there in the request by design
+        Optional<ReadShareGroupStateSummaryResponseData> error = maybeGetReadStateSummaryError(request, offset);
+        if (error.isPresent()) {
+            return error.get();
+        }
+
+        Uuid topicId = request.topics().get(0).topicId();
+        int partition = request.topics().get(0).partitions().get(0).partition();
+
+        SharePartitionKey coordinatorKey = SharePartitionKey.getInstance(request.groupId(), topicId, partition);
+
+        if (!shareStateMap.containsKey(coordinatorKey)) {
+            return ReadShareGroupStateSummaryResponse.toResponseData(
+                    topicId,
+                    partition,
+                    PartitionFactory.UNINITIALIZED_START_OFFSET,
+                    PartitionFactory.DEFAULT_STATE_EPOCH
+            );
+        }
+
+        ShareGroupOffset offsetValue = shareStateMap.get(coordinatorKey, offset);
+
+        if (offsetValue == null) {
+            // Returning an error response as the snapshot value was not found
+            return ReadShareGroupStateSummaryResponse.toErrorResponseData(
+                    topicId,
+                    partition,
+                    Errors.UNKNOWN_SERVER_ERROR,
```

Comment:
Is it the right error or we need to be specific about missing key? Also shouldn't we log in such cases as `unknown server error` is returned which means that this scenario should not occur in broker.